tests/behat/behat-features/src/PushingContext.php

```diff
@@ -0,0 +1,269 @@
+    $cron_link_element = $this->getSession()->getPage()->find('xpath', $xpath);
+    $cron_link = $cron_link_element->getAttribute('href');
+    $client = new Client();
+    $response = $client->request('GET', $cron_link);
+    $response_code = $response->getStatusCode();
+    if ($response_code != 204) {
+      throw new ExpectationException('Expected response code 204 when running cron via link but got: ' . $response_code, $this->getSession());
+    }
+  }
+
+  /**
+   * Waits for x milliseconds.
+   *
+   * @Given I wait for :milliseconds milliseconds
+   */
+  public function waitForSomeTime($milliseconds) {
+    echo $this->getSession()->getCurrentUrl();
+    $this->getSession()->wait((int) $milliseconds);
+  }
+
+  /**
+   * I should see in content overview article with title and random suffix.
+   *
+   * @Then I should see in content overview article with :title and random suffix
+   */
+  public function seeArticleInContentOverviewWithRandomSuffix($title) {
+    $tries = 0;
+    $tries_limit = 5;
+    $title = $title . $this->randomSuffix;
+
+    do {
+      // Sleep one second before next try.
+      if ($tries) {
+        sleep(1);
+      }
+      $this->visitPath('/admin/content');
+      $content = $this->getSession()->getPage()->getContent();
+      $found = strpos($content, $title) !== FALSE;
+    } while (!$found && ++$tries < $tries_limit);
+    if (!$found) {
+      throw new ExpectationException('Article was not found in content overview.', $this->getSession());
+    }
+  }
+
+  /**
+   * I should see current site registered.
+   *
+   * @Then I should see current site registered
+   */
+  public function seeCurrentSiteRegistered() {
+    $tries = 0;
+    $tries_limit = 5;
+    $site_uuid = \Drupal::config('system.site')->get('uuid');
+
+    do {
+      // Sleep one second before next try.
+      if ($tries) {
+        sleep(1);
+      }
+      $this->visitPath('/admin/config/remote-registrations');
+      $content = $this->getSession()->getPage()->getContent();
+      $found = strpos($content, $site_uuid) !== FALSE;
+    } while (!$found && ++$tries < $tries_limit);
+    if (!$found) {
+      throw new ExpectationException('Site uuid was not found in remote registration.', $this->getSession());
+    }
+  }
+
+  /**
+   * Check if this is contentpool site.
+   *
+   * @When I am on contentpool
+   */
+  public function iAmOnContentpool() {
+    $url = $this->getSession()->getCurrentUrl();
+    $url_parts = parse_url($url);
+    if (!isset($url_parts['host']) || strpos($url_parts['host'], 'contentpool') === FALSE) {
+      throw new ExpectationException('Expected contentpool url, got: ' . $url, $this->getSession());
+    }
+  }
+
+  /**
+   * Check if this is satellite site.
+   *
+   * @When I am on satellite
+   */
+  public function iAmOnSatellite() {
+    $url = $this->getSession()->getCurrentUrl();
+    $url_parts = parse_url($url);
+    if (!isset($url_parts['host']) || strpos($url_parts['host'], 'satellite') === FALSE) {
```

Comment:
tests should work with every URLs - so this should compare the host with the respective environment variable